feat: complete events.Event docs (#35360)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/events/event/addrules/index.md

---
title: Event.addRules()
slug: Mozilla/Add-ons/WebExtensions/API/events/Event/addRules
page-type: webextension-api-event
browser-compat: webextensions.api.events.Event.addRules
---

{{AddonSidebar}}

Registers rules to handle events.

## Syntax

```js-nolint
events.Event.addRules(rules, callback)
```

### Parameters

- `rules`

>   - : `array` of `events.Rule`. Specify the rules to be registered. Rules here do not replace previously registered rules.

Reviewer: ```suggestion
  - : `array` of {{WebExtAPIRef("events.Rule")}}. Rules to register. These rules do not replace the registered rules.
```

- `callback` {{optional_inline}}

>   - : `function`. Called when the rules are registered. The parameter of the callback function is an array of `events.Rule` objects that were successfully registered.

Reviewer: ```suggestion
  - : `function`. Called when the rules are registered. The parameter of the callback function is an array of the {{WebExtAPIRef("events.Rule")}} objects registered.
```
